sustainml_cpp/test/blackbox/api/ManagedNodeService.hpp

```diff
@@ -0,0 +1,177 @@
+// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file ManagedNodeService.hpp
+ *
+ */
+
+#ifndef _TEST_BLACKBOX_MANAGEDNODESERVICE_HPP_
+#define _TEST_BLACKBOX_MANAGEDNODESERVICE_HPP_
+
+#include <future>
+
+#include <sustainml_cpp/nodes/AppRequirementsNode.hpp>
+#include <sustainml_cpp/nodes/CarbonFootprintNode.hpp>
+#include <sustainml_cpp/nodes/HardwareConstraintsNode.hpp>
+#include <sustainml_cpp/nodes/HardwareResourcesNode.hpp>
+#include <sustainml_cpp/nodes/MLModelNode.hpp>
+#include <sustainml_cpp/nodes/MLModelMetadataNode.hpp>
+#include <sustainml_cpp/core/RequestReplyListener.hpp>
+
+template <typename _NODE_TYPE, typename _LISTENER_TYPE, typename ... Args>
+class ManagedNodeService
```

Comment:
Instead of maintaining this new `ManagedNodeService` class and the old `ManagedNode` , extend the old `ManagedNode` including the logic of this new one (which is basically adding the `ManagedRequestReplyListener`)